src/Symfony/Component/Config/ValidatorConfigCacheFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config;
+
+/**
+ * A ConfigCacheFactory implementation that validates the
+ * cache with an arbitrary set of metadata validators.
+ *
+ * @author Matthias Pigulla <mp@webfactory.de>
+ */
+class ValidatorConfigCacheFactory implements ConfigCacheFactoryInterface
```

Comment:
For BC reasons, the `Router` and `Translator` need to create a `ConfigCacheFactory` themselves if none is injected. If that factory comes without the default `ResourceInterfaceValidator`, they'd have to add one themselves and that would cause unnecessary dependencies. Having a default `ResourceInterfaceValidator` in `ConfigCacheFactory` would be weird as well.

So this way, use a `ConfigCacheFactory` if you don't care about special validations and you'll get the same behaviour as in the past.

Use a ValidatorConfigCacheFactory if you care about special validations or if you get the instance injected anyway and don't care about the hassle of setting it up.